packages/lesswrong/lib/utils/schemaUtils.ts

```diff
@@ -394,3 +394,35 @@ export function googleLocationToMongoLocation(gmaps: AnyBecauseTodo) {
     coordinates: [gmaps.geometry.location.lng, gmaps.geometry.location.lat]
   }
 }
+export function schemaDefaultValue<T extends DbObject>(defaultValue: any): Partial<CollectionFieldSpecification<T>> {
+  // Used for both onCreate and onUpdate
+  const fillIfMissing = ({ newDocument, fieldName }: {
+    newDocument: T;
+    fieldName: string;
+  }) => {
+    if (newDocument[fieldName as keyof T] === undefined) {
+      return defaultValue instanceof DeferredForumSelect ? defaultValue.get() : defaultValue;
+    } else {
+      return undefined;
+    }
+  };
+  const throwIfSetToNull = ({ oldDocument, document, fieldName }: {
+    oldDocument: T;
+    document: T;
+    fieldName: string;
+  }) => {
+    const wasValid = (oldDocument[fieldName as keyof T] !== undefined && oldDocument[fieldName as keyof T] !== null);
+    const isValid = (document[fieldName as keyof T] !== undefined && document[fieldName as keyof T] !== null);
+    if (wasValid && !isValid) {
+      throw new Error(`Error updating: ${fieldName} cannot be null or missing`);
+    }
+  };
+
+  return {
+    defaultValue: defaultValue,
+    onCreate: fillIfMissing,
+    onUpdate: throwIfSetToNull,
+    canAutofillDefault: true,
+    nullable: false
```

Comment:
Note for other reviewers: we now set `nullable: false` by default when using `schemaDefaultValue`.  This can still be overriden by explicitly setting `nullable: true` after spreading the `schemaDefaultValue` output in a field definition, if you want to explicitly allow `null` values for that column.